BSPConversionLib/Source/EntityConverter.cs

```diff
@@ -137,6 +143,88 @@ private HashSet<string> GetGiveTargets()
 			return targets;
 		}
 
+		private void ConvertFuncDoor(Entity entity)
+		{
+			GetMoveDir(entity);
+
+			if (float.TryParse(entity["health"], out var health))
+			{
+				entity.ClassName = "func_button"; // Health is obsolete on func_door, maybe fix in engine and update this
+			}
+		}
+
+		private void ConvertFuncButton(Entity entity)
+		{
+			GetMoveDir(entity);
+
+			GetButtonFlags(entity);
+
+			if (entity["wait"] == "-1")
+			{
+				entity["wait"] = "0.001";
+			}
+
+			var targets = GetTargetEntities(entity);
+
+			foreach (var target in targets)
+			{
+				switch (target.ClassName)
+				{
+					case "func_door":
+						ButtonDoorLink(entity, target);
+						break;
+				}
+			}
+		}
+
+		private static void ButtonDoorLink(Entity entity, Entity target)
+		{
+			var connection = new Entity.EntityConnection()
+			{
+				name = "OnPressed",
+				target = target["targetname"],
+				action = "Open",
+				param = null,
+				delay = 0,
+				fireOnce = -1
+			};
+			entity.connections.Add(connection);
+		}
+
+		private static void GetButtonFlags(Entity entity)
```

Comment:
This method should be called `SetButtonFlags`. Also `entity` should be renamed to `button`.